contracts/SpokePool.sol

```diff
@@ -1,16 +1,161 @@
+    event FundsDeposited(
+        uint256 nonce,
+        address originToken,
+        uint256 destinationChainId,
+        address recipient,
+        address sender,
+        address destinationToken,
+        uint256 amount,
+        uint64 relayerFeePct,
+        uint64 quoteTimestamp
+    );
+
+    constructor(
+        address timerAddress
+    ) Testable(timerAddress) {
+        deploymentTime = uint64(getCurrentTime());
+    }
+
+    /****************************************
+     *               MODIFIERS              *
+     ****************************************/
+
+    modifier onlyIfDepositsEnabled(address originToken, uint256 destinationId) {
+        require(whitelistedDestinationRoutes[originToken][destinationId].depositsEnabled, "Deposits disabled");
+        _;
+    }
+
+    /**************************************
+     *          ADMIN FUNCTIONS           *
+     **************************************/
+
+    /**
+     * @notice Whitelist an origin token <-> destination token route.
+     */
+    function _whitelistRoute(
+        address originToken,
+        address destinationToken,
+        address spokePool,
+        address wethContract,
+        uint256 destinationChainId
+    ) internal {
+        whitelistedDestinationRoutes[originToken][destinationChainId] = DestinationToken({
+            token: destinationToken,
+            spokePool: spokePool, // Depositing to a destination chain where spoke pool is the 0 address will fail,
+            // so admin can set `spokePool` to 0 address to block deposits.
+            wethContract: wethContract,
+            depositsEnabled: true
+        });
+
+        emit WhitelistRoute(originToken, destinationChainId, destinationToken, spokePool, wethContract);
+    }
+
+    /**
+     * @notice Enable/disable deposits for a whitelisted origin token.
+     */
+    function _setEnableDeposits(address originToken, uint256 destinationChainId, bool depositsEnabled) internal {
+        whitelistedDestinationRoutes[originToken][destinationChainId].depositsEnabled = depositsEnabled;
+        emit DepositsEnabled(originToken, destinationChainId, depositsEnabled);
+    }
+
+    /**************************************
+     *         DEPOSITOR FUNCTIONS        *
+     **************************************/
+
+    /**
+     * @notice Called by user to bridge funds from origin to destination chain.
+     * @dev The caller must first approve this contract to spend `amount` of `originToken`.
+     */
     function deposit(
         address originToken,
+        uint256 destinationChainId,
         uint256 amount,
         address recipient,
-        uint256 destinationChainId,
-        uint256 relayerFee
-    ) public {}
+        uint64 relayerFeePct,
+        uint64 quoteTimestamp
+    ) public onlyIfDepositsEnabled(originToken, destinationChainId) {
+        require(isWhitelistedRoute(originToken, destinationChainId), "deposit token not whitelisted");
```

Comment:
can this be rolled into the modifier above? or should it be it's own modifier?

Reply:
On second thought, these kind of check the same thing. I'm going to remove  the method `isWhitelistedRoute`, which i copied from Across V1.